File: go/day-1/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"path/filepath"
	"sort"
	"strconv"
)

func main() {

	f, _ := os.Open(filepath.Base("input.txt"))

	fileScanner := bufio.NewScanner(f)

	calories := []int{}
	var currentSum int

	for fileScanner.Scan() {

		line := fileScanner.Text()

		if len(line) == 0 {
			calories = append(calories, currentSum)
			currentSum = 0
			continue
		}

		lineCalories, _ := strconv.Atoi(line)

		currentSum += lineCalories
	}

	calories = append(calories, currentSum)

	sort.Ints(calories)

	fmt.Println(calories[len(calories)-1])

}

File: python/day-6/part1.py
text = ""

with open("input.txt") as f:
    text = f.readline()

currentIndex = 3

letters = [*text[:4]]

while len(set(letters)) < 4:
    currentIndex += 1

    letters.pop(0)
    letters.append(text[currentIndex])

print(currentIndex + 1)
